tests/unit/test_parser_whitespace.py

```diff
@@ -0,0 +1,111 @@
+#
+# This file is part of apacheconfig software.
+#
+# Copyright (c) 2018-2019, Ilya Etingof <etingof@gmail.com>
+# License: https://github.com/etingof/apacheconfig/LICENSE.rst
+#
+import sys
+
+from apacheconfig import *
+
+try:
+    import unittest2 as unittest
+
+except ImportError:
+    import unittest
+
+
+def _create_parser(options={}, start='contents'):
+    ApacheConfigLexer = make_lexer(**options)
+    ApacheConfigParser = make_parser(**options)
+    return ApacheConfigParser(ApacheConfigLexer(), start=start)
+
+class WhitespaceParserTestCase(unittest.TestCase):
+    def _test_cases(self, cases, tag='contents', options={}):
+        options['preserve_whitespace'] = True
+        parser = _create_parser(start=tag, options=options)
+        for x, expected in cases:
```

Comment:
When we drop Py2 support, [subtests](https://docs.python.org/3/library/unittest.html#subtests) perhaps would be helpful here.

Reply:
yeah! I created an issue to track this: #57